daemon/log.c

```diff
@@ -253,14 +253,14 @@ void log_message(MTC_S32 priority, PMTC_S8 fmt, ...)
         return;
     }
 
+    if (priority & MTC_LOG_PRIVATELOG && fpLogfile != NULL && privatelogflag)
```

Comment:
I've just realised there is a potential issue here, as `log_reopen()` could have closed `fpLogfile`, then failed to open the new file, leaving it NULL after we check it here, so I probably need to move the test of fpLogfile to be protected by the lock...